configuration_go/k8sutil/v2/builders.go

```diff
@@ -0,0 +1,132 @@
+package v2
+
+import (
+	"log"
+
+	"github.com/observatorium/observatorium/configuration_go/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	defaultTerminationGracePeriodSeconds = 120
+	ServiceManifestKey                   = "service"
+	ServiceMonitorManifestKey            = "serviceMonitor"
+	ServiceAccountManifestKey            = "serviceAccount"
+	StatefulSetManifestKey               = "statefulSet"
+)
+
+type StatefulSetBuilder struct {
+	MetaConfig                    MetaConfig
+	Containers                    []ContainerProvider
+	Replicas                      int32
+	TerminationGracePeriodSeconds int64
+	Affinity                      *corev1.Affinity
+	SecurityContext               *corev1.PodSecurityContext
+	ServiceAccountName            string
+}
+
+func (s *StatefulSetBuilder) makeAffinity() *corev1.Affinity {
```

Comment:
Any reason to keep these methods private?

Reply:
It is not meant to be called by the user. If the user wants a custom affinity, he can set the attribute directly. This is just code organisation. I can push them at the bottom.

Reply:
Removed the builder